app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,38 @@
+class LikesController < ApiController
+  before_action :set_like, only: [:destroy]
+
+  # GET /likes
+  def index
+    likes = Like.all
+    render :json => likes.to_json
+  end
+
+  # POST /likes
+  def create
+    content = like_params[:content_type].constantize.find(like_params[:content_id])
+    like = content.likes.new(like_params)
+    if like.save!
+      render :json => like.to_json, status: :created
+    else
+      render :json => like.errors, status: :unprocessable_entity
+    end
+  rescue NoMethodError, NameError, ActiveRecord::RecordNotFound
+    head :unprocessable_entity
+  end
+
+  # DELETE /likes/1
+  def destroy
+    @like.destroy
+    render json: 'Like was successfully destroyed.', status: :ok
+  end
+
+  private
+
+  def set_like
+    @like = Like.find(params[:id])
```

Comment:
If this was changed to `Like.find_by(like_params)` and the routes where changed so that you had to hit `DELETE /likes` with all of those parameters in order to delete a like it would work without adding the `id` to the table.

Reply:
Thanks, I'll try that. I think I had something similar at first. I'll update here.

Reply:
So I tried with this approach, but ActiveRecord doesn't really like models not having an `id`. Essentially it's impossible now in Rails 5 to delete an instance without an id or some other primary key.

The issue with setting the three columns on this table as a compound primary key is that FactoryGirl doesn't seem to like that setup and won't build the object.

I get that this could function almost as a join table, since no record should be duplicated, even with different `id`s, but I don't think it's worth the hurdles we have to jump through.